### PR TITLE
Prac 08 feedback

### PR DESCRIPTION
**Pull request by AdriantoHartono**
Hi, can you guys help me to check the code?😅
@Lijiayi2
@JevanChandra
@a-rosa
@nora-zyy
@Riston-gitting-gud
@LiJianhong-qwq
@MikeLeongZhiFai

### CODE REVIEW

File: prac_08/box_layout.kv [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        text: ''
+
+    Button:
+        text: 'Greet'
+        on_press: app.handle_greet()
+
+    Label:
+        id: output_label
+        text: 'Enter your name'
+        color: (1, 1, 0, 1)
```
Comment (Lijiayi2) [inline prac_08/box_layout.kv:23]: good code

File: prac_08/box_layout_demo.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    def handle_greet(self):
+        print("test")
+        self.root.ids.output_label.text = f"Hello {self.root.ids.input_name.text}"
+
+    def clear_input(self):
+        self.root.ids.input_name.text = f""
+        self.root.ids.output_label.text = "Enter your name"
+
+BoxLayoutDemo().run()
```
Comment (Lijiayi2) [inline prac_08/box_layout_demo.py:21]: I think no problem

File: prac_08/convert_miles_km.kv [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    Button:
+        text: "Convert miles to km"
+        font_size: 60
+        on_press: app.calculate_value()
+
+    Label:
+        id: output_kilometer
+        font_size: 60
+        text: " "
```
Comment (Lijiayi2) [inline prac_08/convert_miles_km.kv:36]: I think that you should add number

File: prac_08/convert_miles_km.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+            return 0
+
+    def calculate_value(self):
+        """Calculating value from miles to kilometer"""
+        value = self.get_valid_value()
+        result = value * MILES_TO_KM
+        self.root.ids.output_kilometer.text = str(result)
+
+
+ConvertMilesToKilometer().run()
```
Comment (Lijiayi2) [inline prac_08/convert_miles_km.py:40]: it's ok

File: prac_08/dynamic_labels.kv [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#Adrianto Hartono
+#14347930
+
+BoxLayout:
+    BoxLayout:
+        orientation:"vertical"
+        id: main
```
Comment (Lijiayi2) [inline prac_08/dynamic_labels.kv:7]: I think you probably didn't finish

File: prac_08/dynamic_labels.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        return self.root
+
+    def create_labels(self):
+        """Create labels from data and add them to the GUI"""
+        for name in NAMES_LIST:
+            temp_label = Label(text=name)
+            self.root.ids.main.add_widget(temp_label)
+
+
+DynamicLabelApp().run()
```
Comment (Lijiayi2) [inline prac_08/dynamic_labels.py:28]: good

File: prac_08/squaring.kv [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+            # the following line specifies the function in the app class to call when the button is pressed
+            on_press: app.handle_calculate(input_number.text)
+
+        Label:
+            id: output_label
+            font_size: 48
+            color: (1, 0.7, 0.7, 1)
+
+    Label:
+        text: 'Enter a number and press "Square"'
```
Comment (Lijiayi2) [inline prac_08/squaring.kv:27]: good code

File: prac_08/squaring.py [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    def handle_calculate(self, value):
+        """ handle calculation (could be button press or other call), output result to label widget """
+        try:
+            result = float(value) ** 2
+            self.root.ids.output_label.text = str(result)
+        except ValueError:
+            pass
+
+
+SquareNumberApp().run()
```
Comment (Lijiayi2) [inline prac_08/squaring.py:36]: Can run ALL good

File: prac_08/box_layout.kv
Comment (JevanChandra) [inline prac_08/box_layout.kv:1]: Very clean layout!

File: prac_08/box_layout_demo.py
Comment (JevanChandra) [inline prac_08/box_layout_demo.py:1]: Nice code!

File: prac_08/convert_miles_km.kv
Comment (JevanChandra) [inline prac_08/convert_miles_km.kv:1]: Good Layout

File: prac_08/convert_miles_km.py
Comment (JevanChandra) [inline prac_08/convert_miles_km.py:1]: Great Code

File: prac_08/dynamic_labels.py
Comment (JevanChandra) [inline prac_08/dynamic_labels.py:1]: Good Job!

Comment (MikeLeongZhiFai) [reply prac_08/dynamic_labels.py:1]: Great code! It worked well as expected

File: prac_08/box_layout_demo.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        self.title = "Box Layout Demo"
+        self.root = Builder.load_file('box_layout.kv')
+        return self.root
+
+    def handle_greet(self):
+        print("test")
+        self.root.ids.output_label.text = f"Hello {self.root.ids.input_name.text}"
+
+    def clear_input(self):
+        self.root.ids.input_name.text = f""
```
Comment (a-rosa) [inline prac_08/box_layout_demo.py:18]: You don't need to have the f in front of the ""

File: prac_08/box_layout.kv
Comment (AdriantoHartono) [reply prac_08/box_layout.kv:1]: Thank you🙏

File: prac_08/box_layout.kv (hunk 1 above)
Comment (AdriantoHartono) [reply prac_08/box_layout.kv:23]: Thank you🙏

File: prac_08/box_layout_demo.py
Comment (AdriantoHartono) [reply prac_08/box_layout_demo.py:1]: Thank you🙏

File: prac_08/box_layout_demo.py (hunk 9 above)
Comment (AdriantoHartono) [reply prac_08/box_layout_demo.py:18]: Thank you for the feedback, noteed🙏

File: prac_08/box_layout_demo.py (hunk 2 above)
Comment (AdriantoHartono) [reply prac_08/box_layout_demo.py:21]: Thank you🙏

File: prac_08/convert_miles_km.kv
Comment (AdriantoHartono) [reply prac_08/convert_miles_km.kv:1]: Thank you🙏

File: prac_08/convert_miles_km.kv (hunk 3 above)
Comment (AdriantoHartono) [reply prac_08/convert_miles_km.kv:36]: Thank you for the feedback, noted🙏

File: prac_08/convert_miles_km.py
Comment (AdriantoHartono) [reply prac_08/convert_miles_km.py:1]: Thank you🙏

File: prac_08/convert_miles_km.py (hunk 4 above)
Comment (AdriantoHartono) [reply prac_08/convert_miles_km.py:40]: Thank you🙏

File: prac_08/dynamic_labels.kv (hunk 5 above)
Comment (AdriantoHartono) [reply prac_08/dynamic_labels.kv:7]: Thank you for the feedback, but that is all for the code🙏

File: prac_08/dynamic_labels.py
Comment (AdriantoHartono) [reply prac_08/dynamic_labels.py:1]: Thank you🙏

File: prac_08/dynamic_labels.py (hunk 6 above)
Comment (AdriantoHartono) [reply prac_08/dynamic_labels.py:28]: Thank you🙏

File: prac_08/squaring.kv (hunk 7 above)
Comment (AdriantoHartono) [reply prac_08/squaring.kv:27]: Thank you🙏

File: prac_08/squaring.py (hunk 8 above)
Comment (AdriantoHartono) [reply prac_08/squaring.py:36]: Thank you🙏

File: prac_08/dynamic_labels.py (hunk 6 above)
Comment (nora-zyy) [reply prac_08/dynamic_labels.py:28]: Brief and clear, easy to understand.

**Pull request merged**